scripts/profiler/data_collector.py

```diff
@@ -20,23 +52,25 @@ def main():
     args = parser.parse_args()
 
     if args.log is not None:
-	    log_lvl_number = int(getattr(logging, args.log.upper(), None))
+        log_lvl_number = int(getattr(logging, args.log.upper(), None))
     else:
-	    log_lvl_number = logging.DEBUG
-
-    def sigint_handler(sig, frame):
-        end_ev.set()
-
-    signal.signal(signal.SIGINT, sigint_handler)
-    end_ev = threading.Event()
+        log_lvl_number = logging.INFO
 
-    profiler = RttNordicProfilerHost(
-                event_filename=args.dataset_name + ".csv",
-                finish_event=end_ev,
-                event_types_filename=args.dataset_name + ".json",
-                log_lvl=log_lvl_number)
-    profiler.get_events_descriptions()
-    profiler.read_events_rtt(args.time)
+    try:
+        lock = Lock()
+        p1 = Process(target=model_creator, args=(lock, SOCK_ADDR_DICT, args.dataset_name, log_lvl_number,), daemon=True)
+        p2 = Process(target=rtt2socket, args=(lock, SOCK_ADDR_DICT, log_lvl_number,), daemon=True)
+        # The order of starting processes does matter.
+        p1.start()
```

Comment:
Can we be sure that process that is started as first one will acquire the lock first ?

Reply:
![image](https://user-images.githubusercontent.com/86837740/130193936-9d652f59-1610-4312-ba3d-3700ef6fbf04.png)
https://docs.python.org/3/library/multiprocessing.html
From what I can see here, I believe yes.

Reply:
It seems like it only assures that only one process at a time performs print operation. It still does no assure order I think.